core/commonMain/src/kotlinx/serialization/descriptors/SerialDescriptors.kt

```diff
@@ -223,6 +223,20 @@ public val SerialDescriptor.nullable: SerialDescriptor
         return SerialDescriptorForNullable(this)
     }
 
+/**
+ * Returns non-nullable serial descriptor for the type, if [this] is a nullable descriptor created automatically or
+ * with `.nullable` extensions on a descriptor or serializer. Otherwise, returns [this].
+ *
+ * @see SerialDescriptor.nullable
+ * @see KSerializer.nullable
+ */
+@ExperimentalSerializationApi
+public val SerialDescriptor.unwrapNullable: SerialDescriptor
```

Comment:
After looking at the api dump, I've realized that `unwrap` is actually a verb, and its counter-intuitive to have a verb in property name (https://kotlinlang.org/docs/coding-conventions.html#choose-good-names). So it should be either

1. A function `SerialDescriptor.unwrapNullable()`
or
2. A property with a noun, e.g. `unwrappedNullable`.

I'm not sure what is better. Doing (1) will bring asymmetry with `.nullable`. (2) is confusing, because it is not clear what 'unwrapped' is since we do not have 'wrapped' anywhere else. Besides, 'unwrappedNullable' sounds like some flavor of a nullable thing, but in reality, returned descriptor is non-nullable.

Reply:
You're right, I'm feeling the same. I would prefer `unwrappedNullable` to not have this asymmetry. I'll commit now. If you feel not comfortable, could we have a third person reviewing this to help chosing a better name ?

Reply:
@shanshin or @qwwdfsad may help here, I guess

Reply:
Or, we can do `unwrappedOriginal` and also unwrap `WrappedSerialDescriptor.original` (I've looked at all the SerialDescriptors and only SerialDescriptorForNullable and WrappedSerialDescriptor would have sense of unwrapping the original descriptor)

Reply:
I like the idea, but it would mean also renaming the current data class  `WrappedSerialDescriptor` to something else to allow creation of the interface. What do you think @sandwwraith ?

Reply:
> the route of introducing a WrappedSerialDescriptor interface would make a lot of sense, it would also future proof the design to support other wrapping descriptors

IMO this is way out of scope of this PR.

Reply:
WDYT of `nonNullableOriginal` name? We can add to the documentation that this property "does not affect descriptors that were created as nullable by directly implementing SerialDescriptor interface."

Reply:
New doc, is it okay for you @sandwwraith ?
```

 * Returns non-nullable serial descriptor for the type, only if this descriptor has been auto-generated (plugin
 * generated descriptors) or with `.nullable` extension on a descriptor or serializer.
 *
 * Otherwise, returns this.
 * 
 * It may return nullable descriptor if this descriptor has been created manually as nullable by directly implementing SerialDescriptor interface.
 *
```